diff -uw ehbasic.orig/basic.asm ehbasic/basic.asm
--- ehbasic.orig/basic.asm	2010-07-29 15:58:42.000000000 -0400
+++ ehbasic/basic.asm	2012-04-20 23:20:43.000000000 -0400
@@ -419,7 +419,7 @@
 LAB_SKFF		= LAB_STAK+$FF
 					; flushed stack address
 
-ccflag		= $0200	; BASIC CTRL-C flag, 00 = enabled, 01 = dis
+ccflag		= $0300	; BASIC CTRL-C flag, 00 = enabled, 01 = dis
 ccbyte		= ccflag+1	; BASIC CTRL-C byte
 ccnull		= ccbyte+1	; BASIC CTRL-C byte timeout
 
@@ -436,12 +436,12 @@
 					; start of input buffer after IRQ/NMI code
 Ibuffe		= Ibuffs+$47; end of input buffer
 
-Ram_base		= $0300	; start of user RAM (set as needed, should be page aligned)
-Ram_top		= $C000	; end of user RAM+1 (set as needed, should be page aligned)
+Ram_base		= $0400	; start of user RAM (set as needed, should be page aligned)
+Ram_top		= $5000	; end of user RAM+1 (set as needed, should be page aligned)
 
 ; This start can be changed to suit your system
 
-	*=	$C000
+	.org	$5000
 
 ; BASIC cold start entry point
 
@@ -1555,7 +1555,7 @@
 	JMP	LAB_LET		; else go do implied LET
 
 LAB_1609
-	CMP	#[TK_TAB-$80]*2	; compare normalised token * 2 with TAB
+	CMP	#(TK_TAB-$80)*2	; compare normalised token * 2 with TAB
 	BCS	LAB_15D9		; branch if A>=TAB (do syntax error then warm start)
 					; only tokens before TAB can start a line
 	TAY				; copy to index
Only in ehbasic: Makefile
diff -uw ehbasic.orig/min_mon.asm ehbasic/min_mon.asm
--- ehbasic.orig/min_mon.asm	2005-04-08 02:53:46.000000000 -0400
+++ ehbasic/min_mon.asm	2012-04-20 23:13:11.000000000 -0400
@@ -1,5 +1,6 @@
 
 ; minimal monitor for EhBASIC and 6502 simulator V1.05
+; Modified to support the Replica 1 by Jeff Tranter <tranter@pobox.com>
 
 ; To run EhBASIC on the simulator load and assemble [F7] this file, start the simulator
 ; running [F6] then start the code with the RESET [CTRL][SHIFT]R. Just selecting RUN
@@ -23,7 +24,7 @@
 ; and wait for the user to select [C]old or [W]arm start. nothing else
 ; fits in less than 128 bytes
 
-	*=	$FF80			; pretend this is in a 1/8K ROM
+;	.org	$FF80			; pretend this is in a 1/8K ROM
 
 ; reset vector points here
 
@@ -67,18 +68,22 @@
 LAB_dowarm
 	JMP	LAB_WARM		; do EhBASIC warm start
 
-; byte out to simulated ACIA
+; byte out to Replica 1/Apple 1 screen
 
 ACIAout
-	STA	ACIAsimwr		; save byte to simulated ACIA
+	BIT	$D012
+	BMI	ACIAout
+	STA	$D012
 	RTS
 
-; byte in from simulated ACIA
+; byte in from Replica 1/Apple 1 keyboard
 
 ACIAin
-	LDA	ACIAsimrd		; get byte from simulated ACIA
-	BEQ	LAB_nobyw		; branch if no byte waiting
+	LDA	$D011
+	BPL	LAB_nobyw		; branch if no byte waiting
 
+	LDA	$D010
+	AND	#$7F			; clear high bit
 	SEC				; flag byte received
 	RTS
 
@@ -126,7 +131,7 @@
 
 ; system vectors
 
-	*=	$FFFA
+	.org	$FFFA
 
 	.word	NMI_vec		; NMI vector
 	.word	RES_vec		; RESET vector
Only in ehbasic: NOTES.txt
Only in ehbasic: REPLICA1.patch
Only in ehbasic: source.zip
