diff --git a/asminc/sbc.inc b/asminc/sbc.inc
new file mode 100644
index 000000000..416ad0f27
--- /dev/null
+++ b/asminc/sbc.inc
@@ -0,0 +1,25 @@
+; ---------------------------------------------------------------------------
+;
+; SBC definitions
+;
+; ---------------------------------------------------------------------------
+
+RAMSTART := $0200     ; Entry point
+
+; ---------------------------------------------------------------------------
+; Hardware Addresses
+; ---------------------------------------------------------------------------
+
+ACIA        := $A000
+ACIAControl := ACIA+0
+ACIAStatus  := ACIA+0
+ACIAData    := ACIA+1
+
+; ---------------------------------------------------------------------------
+; Monitor Functions
+; ---------------------------------------------------------------------------
+
+MONITOR  := $DF00     ; Monitor entry point
+MONCOUT  := $FF3B     ; Console out routine
+MONRDKEY := $FF4A     ; Console in routine
+ClearScreen := $EED2  ; Clear screen
diff --git a/cfg/sbc.cfg b/cfg/sbc.cfg
new file mode 100644
index 000000000..0d24bc2ae
--- /dev/null
+++ b/cfg/sbc.cfg
@@ -0,0 +1,39 @@
+# sbc.cfg   (32k)
+#
+# Jeff Tranter's 6502 Single Board Computer
+#
+# ld65 --config sbc.cfg -o <prog>.bin <prog>.o
+
+FEATURES {
+    STARTADDRESS: default = $0200;
+    CONDES: segment = STARTUP,
+            type    = constructor,
+            label   = __CONSTRUCTOR_TABLE__,
+            count   = __CONSTRUCTOR_COUNT__;
+    CONDES: segment = STARTUP,
+            type    = destructor,
+            label   = __DESTRUCTOR_TABLE__,
+            count   = __DESTRUCTOR_COUNT__;
+}
+
+SYMBOLS {
+    __STACKSIZE__:    type = weak, value = $0080;   # 128 byte program stack
+    __STARTADDRESS__: type = export, value = %S;
+}
+
+MEMORY {
+    ZP:       file = %O, define = yes, start = $0000, size = $00EE;
+    CPUSTACK: file = "", define = yes, start = $0100, size = $0100;
+    RAM:      file = %O, define = yes, start = %S, size = $8000 - %S - __STACKSIZE__;
+    ROM:      file = "", define = yes, start = $C000, size = $4000;
+}
+
+SEGMENTS {
+    ZEROPAGE: load = ZP,  type = zp,  define = yes;
+    STARTUP:  load = RAM, type = ro,  define = yes;
+    CODE:     load = RAM, type = ro,  define = yes;
+    RODATA:   load = RAM, type = ro,  define = yes;
+    ONCE:     load = RAM, type = ro,  define = yes;
+    DATA:     load = RAM, type = rw,  define = yes;
+    BSS:      load = RAM, type = bss, define = yes;
+}
diff --git a/libsrc/sbc/clrscr.s b/libsrc/sbc/clrscr.s
new file mode 100644
index 000000000..2afa95b84
--- /dev/null
+++ b/libsrc/sbc/clrscr.s
@@ -0,0 +1,11 @@
+;
+; void clrscr (void);
+;
+
+        .export         _clrscr
+
+        .include        "sbc.inc"
+
+_clrscr:
+        lda     #$0C            ; Form feed
+        jmp     MONCOUT         ; Send character
diff --git a/libsrc/sbc/crt0.s b/libsrc/sbc/crt0.s
new file mode 100644
index 000000000..8703675e2
--- /dev/null
+++ b/libsrc/sbc/crt0.s
@@ -0,0 +1,46 @@
+;
+; Startup code for cc65 (SBC version)
+;
+
+        .export         _init, _exit
+        .export         __STARTUP__ : absolute = 1      ; Mark as startup
+
+        .import         _main
+        .import         initlib, donelib, copydata, zerobss
+        .import         __RAM_START__, __RAM_SIZE__     ; Linker generated
+        .import         __STACKSIZE__                   ; Linker generated
+
+        .include        "zeropage.inc"
+        .include        "sbc.inc"
+
+
+; Place the startup code in a special segment
+
+.segment  "STARTUP"
+
+
+; A little light housekeeping
+
+_init:    cld                          ; Clear decimal mode
+
+; Set cc65 argument stack pointer
+
+          lda     #<(__RAM_START__ + __RAM_SIZE__)
+          sta     sp
+          lda     #>(__RAM_START__ + __RAM_SIZE__)
+          sta     sp+1
+
+; Initialize memory storage
+
+          jsr     zerobss              ; Clear BSS segment
+          jsr     copydata             ; Initialize DATA segment
+          jsr     initlib              ; Run constructors
+
+; Call main()
+
+          jsr     _main
+
+; Back from main (this is also the _exit entry). Jumps to monitor
+; entry point.
+
+_exit:    jmp    MONITOR
diff --git a/libsrc/sbc/ctype.s b/libsrc/sbc/ctype.s
new file mode 100644
index 000000000..1301965eb
--- /dev/null
+++ b/libsrc/sbc/ctype.s
@@ -0,0 +1,5 @@
+; Character specification table.
+;
+; uses the "common" definition
+
+        .include "ctype_common.inc"
diff --git a/libsrc/sbc/kbhit.s b/libsrc/sbc/kbhit.s
new file mode 100644
index 000000000..06915954d
--- /dev/null
+++ b/libsrc/sbc/kbhit.s
@@ -0,0 +1,13 @@
+;
+; unsigned char kbhit (void);
+;
+
+                .export _kbhit
+
+.include        "sbc.inc"
+
+_kbhit:
+        lda     ACIAStatus      ; Get ACIA status
+        and     #$01            ; Look at bit 0 - 0 means no data, 1 means data ready
+        ldx     #0              ; Clear MSB of return value
+        rts                     ; Return
diff --git a/libsrc/sbc/read.s b/libsrc/sbc/read.s
new file mode 100644
index 000000000..e3339ead5
--- /dev/null
+++ b/libsrc/sbc/read.s
@@ -0,0 +1,49 @@
+;
+; int __fastcall__ read (int fd, void* buf, unsigned count);
+;
+
+.include        "sbc.inc"
+
+.import         popax, popptr1
+.importzp       ptr1, ptr2, ptr3
+
+.export         _read
+
+.proc           _read
+
+        sta     ptr3
+        stx     ptr3+1           ; Count in ptr3
+        inx
+        stx     ptr2+1           ; Increment and store in ptr2
+        tax
+        inx
+        stx     ptr2
+        jsr     popptr1          ; Buffer address in ptr1
+        jsr     popax
+
+begin:  dec     ptr2
+        bne     getch
+        dec     ptr2+1
+        beq     done             ; If buffer full, return
+
+getch:  jsr     MONRDKEY         ; Get character using Monitor ROM call
+        bcc     getch
+        jsr     MONCOUT          ; Echo it
+        and     #$7F             ; Clear top bit
+        cmp     #$0D             ; Check for '\r'
+        bne     putch            ; ...if CR character
+        lda     #$0A             ; Replace with '\n'
+        jsr     MONCOUT          ; and echo it
+
+putch:  ldy     #$00             ; Put char into return buffer
+        sta     (ptr1),y
+        inc     ptr1             ; Increment pointer
+        bne     begin
+        inc     ptr1+1
+        bne     begin
+
+done:   lda     ptr3
+        ldx     ptr3+1
+        rts                      ; Return count
+
+.endproc
diff --git a/libsrc/sbc/write.s b/libsrc/sbc/write.s
new file mode 100644
index 000000000..fabfdc1a6
--- /dev/null
+++ b/libsrc/sbc/write.s
@@ -0,0 +1,46 @@
+;
+; int __fastcall__ write (int fd, const void* buf, int count);
+;
+
+.include        "sbc.inc"
+
+.import         popax, popptr1
+.importzp       ptr1, ptr2, ptr3
+
+.export         _write
+
+.proc           _write
+
+        sta     ptr3
+        stx     ptr3+1          ; Count in ptr3
+        inx
+        stx     ptr2+1          ; Increment and store in ptr2
+        tax
+        inx
+        stx     ptr2
+        jsr     popptr1         ; Buffer address in ptr1
+        jsr     popax
+
+begin:  dec     ptr2
+        bne     outch
+        dec     ptr2+1
+        beq     done
+
+outch:  ldy     #0
+        lda     (ptr1),y
+        jsr     MONCOUT         ; Send character using Monitor call
+        cmp     #$0A            ; Check for 'n'
+        bne     next            ; ...if LF character
+        lda     #$0D            ; Add a carriage return
+        jsr     MONCOUT
+
+next:   inc     ptr1
+        bne     begin
+        inc     ptr1+1
+        jmp     begin
+
+done:   lda     ptr3
+        ldx     ptr3+1
+        rts                     ; Return count
+
+.endproc
diff --git a/src/ca65/main.c b/src/ca65/main.c
index 3ec6c84ee..5cfdcb562 100644
--- a/src/ca65/main.c
+++ b/src/ca65/main.c
@@ -350,6 +350,10 @@ static void SetSys (const char* Sys)
             NewSymbol ("__RP6502__", 1);
             break;
 
+        case TGT_SBC:
+            NewSymbol ("__SBC__", 1);
+            break;
+
         default:
             AbEnd ("Invalid target name: '%s'", Sys);
 
diff --git a/src/cc65/main.c b/src/cc65/main.c
index bef646cdd..95c9ad07b 100644
--- a/src/cc65/main.c
+++ b/src/cc65/main.c
@@ -307,6 +307,10 @@ static void SetSys (const char* Sys)
             DefineNumericMacro ("__RP6502__", 1);
             break;
 
+        case TGT_SBC:
+            DefineNumericMacro ("__SBC__", 1);
+            break;
+
         default:
             AbEnd ("Unknown target system '%s'", Sys);
     }
diff --git a/src/common/target.c b/src/common/target.c
index b50478e16..dee5110bf 100644
--- a/src/common/target.c
+++ b/src/common/target.c
@@ -174,6 +174,7 @@ static const TargetEntry TargetMap[] = {
     {   "pet",          TGT_PET          },
     {   "plus4",        TGT_PLUS4        },
     {   "rp6502",       TGT_RP6502       },
+    {   "sbc",          TGT_SBC          },
     {   "sim6502",      TGT_SIM6502      },
     {   "sim65c02",     TGT_SIM65C02     },
     {   "supervision",  TGT_SUPERVISION  },
@@ -223,6 +224,7 @@ static const TargetProperties PropertyTable[TGT_COUNT] = {
     { "sym1",           CPU_6502,       BINFMT_BINARY,      CTNone  },
     { "kim1",           CPU_6502,       BINFMT_BINARY,      CTNone  },
     { "rp6502",         CPU_65C02,      BINFMT_BINARY,      CTNone  },
+    { "sbc",            CPU_6502,       BINFMT_BINARY,      CTNone  },
 };
 
 /* Target system */
diff --git a/src/common/target.h b/src/common/target.h
index 730b8211e..06097326b 100644
--- a/src/common/target.h
+++ b/src/common/target.h
@@ -89,6 +89,7 @@ typedef enum {
     TGT_SYM1,
     TGT_KIM1,
     TGT_RP6502,
+    TGT_SBC,
     TGT_COUNT                   /* Number of target systems */
 } target_t;
 
diff --git a/targettest/Makefile b/targettest/Makefile
index 4d989d0df..38800b246 100644
--- a/targettest/Makefile
+++ b/targettest/Makefile
@@ -761,6 +761,7 @@ TARGETS :=     \
   pce          \
   pet          \
   plus4        \
+  sbc          \
   sim6502      \
   sim65c02     \
   supervision  \
